azerothcore/azerothcore-wotlk
# Adjust wander distance and movement type on Splintered Skeleton

## Changes Proposed:
-  Updated Splintered Skeleton wander_distance and MovementType
#### Edit:
- Adjusted the spawn points for Splintered Skeleton according to suggested sniff.
- Added SAI for Hate to zero with 25% chance on Splintered Skeleton.

## Issues Addressed:
https://github.com/azerothcore/azerothcore-wotlk/issues/4520
- Closes 5. Splintered skeletons all aggro together



## SOURCE:

https://youtu.be/s-2GXud7CJI?t=1748 


## Tests Performed:

- Currently Splintered Skeleton remain stationary until aggroed.
![2021-04-11 14-28-58 (1)](https://user-images.githubusercontent.com/10816895/114303406-aa8d5500-9ad6-11eb-91a9-a50cd2ba5a91.gif)
- Tested in game on linux/docker setup from windows client.
- With the changes Splintered Skeleton is wandering in about 3yard radius.
![2021-04-11 14-25-29 (1)](https://user-images.githubusercontent.com/10816895/114303422-b4af5380-9ad6-11eb-8278-1e17815e4acc.gif)
Hate to zero test:
![2021-04-12 21-24-51](https://user-images.githubusercontent.com/10816895/114452820-8d917880-9be1-11eb-98e1-8cef78c16a79.gif)

## How to Test the Changes:

Wandering:
1. Go to Scholomance.
2. Go to Vault of the Ravenian
3. Check if Splintered Skeleton is moving.
4. Apply the DB query.
5. Go to Vault of the Ravenian
6. Check if Splintered Skeleton is now wandering.

Hate to zero:
1. Make sure you have a way to measure threat. (I used 'omen' threat meter for WOTLK)
2. Go to Scholomance.
3. Go to Vault of the Ravenian
4. Check if Splintered Skeleton is moving.
5. Apply the DB query.
6. Go to Vault of the Ravenian
7. Aggro Splintered Skeleton wait n amount of time until hate to zero procs(25% chance). When it procs notice if your threat has been reset.

## Known Issues and TODO List:



## Target Branch(es):







 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
